### PR TITLE
Fix #1510: Handle spacing and links in content box messages.

### LINKED ISSUE #1510: QRScanDialog$WebcamDependencyException: Missing Dependency Error Doesn't Display Link
On Windows 11 Home v23H2, this error references a link, but the link is not displayed. 

image

sparrow.log:

```
2024-09-29 12:14:38,299 ERROR [JavaFX Application Thread] c.s.s.c.FileImportPane [null:-1] Error importing QR
com.sparrowwallet.sparrow.control.QRScanDialog$WebcamDependencyException: Your system is missing a dependency required for the webcam. Follow the link below for more details.

[https://sparrowwallet.com/docs/faq.html#your-system-is-missing-a-dependency-for-the-webcam]
```

source: 
https://github.com/sparrowwallet/sparrow/blob/1bbc586cd616e46c6ebd9615fb0e616ae44fb98c/src/main/java/com/sparrowwallet/sparrow/control/QRScanDialog.java#L143

I was able to fix the problem by installing the security package linked from that FAQ. 

I think this is worth investigating, because some users may be comfortable with downloading and installing a Microsoft security patch (thus working through this independently), but not tracking down the link in source code. 

I'd be interested in working on a PR, but not sure what the contribution policy is. If that's agreeable, just let me know :)

### PR DESCRIPTION
**Pull request by dcavacec**
#1510 
Updates getContentBox to create Hyperlinks for URLs found in the passed message. Two helper functions were added for creating the Hyperlinks and Labels (for non-link text).

### CODE REVIEW

File: src/main/java/com/sparrowwallet/sparrow/control/TitledDescriptionPane.java [hunk 1]
```diff
@@ -127,25 +127,44 @@ protected void setError(String title, String detail) {
...
-        Label details = new Label(message);
-        details.setWrapText(true);
-
-        HBox contentBox = new HBox();
+        // Create the VBox to hold text and Hyperlink components
+        VBox contentBox = new VBox();
         contentBox.setAlignment(Pos.TOP_LEFT);
-        contentBox.getChildren().add(details);
         contentBox.setPadding(new Insets(10, 30, 10, 30));
+        contentBox.setPrefWidth(400);  // Set preferred width for wrapping
```
Comment (craigraw) [inline src/main/java/com/sparrowwallet/sparrow/control/TitledDescriptionPane.java:134]: Please also set `contentBox.setMinHeight(60);`

File: src/main/java/com/sparrowwallet/sparrow/control/TitledDescriptionPane.java [hunk 2]
```diff
@@ -178,4 +197,27 @@ protected static int getAccount(Wallet wallet, KeyDerivation requiredDerivation)
...
+        Hyperlink hyperlink = new Hyperlink(url);
+        hyperlink.setMaxWidth(400);  // Set maximum width for wrapping
+        hyperlink.setWrapText(true);  // Ensure text wrapping in the hyperlink
+        hyperlink.setOnAction(e -> {
+            try {
+                java.awt.Desktop.getDesktop().browse(new java.net.URI(url));
+            } catch (Exception ex) {
+               // TODO: handle failure to launch link
+            }
+        });
```
Comment (craigraw) [inline src/main/java/com/sparrowwallet/sparrow/control/TitledDescriptionPane.java:220]: Please change to `hyperlink.setOnAction(_ -> AppServices.get().getApplication().getHostServices().showDocument(url));`

**Comment by craigraw** [comment]
Thanks - I've added a couple of comments, but it seems to be working well on macOS. I will test on Linux.

**Comment by dcavacec** [comment]
> Thanks - I've added a couple of comments, but it seems to be working well on macOS. I will test on Linux.

Updated!

**Comment by craigraw** [comment]
Tested in Linux, working well. Thanks!

**Pull request merged**